x/gov/keeper/abci.go

```diff
@@ -75,11 +76,18 @@ func (k Keeper) EndBlocker(ctx context.Context) error {
 			return err
 		}
 
+		var res consensustypes.QueryParamsResponse
+
+		if err := k.RouterService.QueryRouterService().InvokeTyped(ctx, &consensustypes.QueryParamsRequest{}, &res); err != nil {
+			return err
+		}
+
 		// called when proposal become inactive
 		// call hook when proposal become inactive
-		if err := k.BranchService.Execute(ctx, func(ctx context.Context) error {
+		_, err = k.BranchService.ExecuteWithGasLimit(ctx, uint64(res.Params.Block.MaxGas), func(ctx context.Context) error {
```

Comment:
Ohh, it's just block gas, it isn't the remaininer of the block gas after the txs in the block have been executed, then my concerns are lifted.

Reply:
Btw, this isn't where the proposal is executed, this is about its hook.

Reply:
Thank you will amend. Haven't completed wanted to do testing.